examples/Perl/train.pl

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/perl
+#
+# Uses KITT.AI RESTful API to train a personal model 
+# Uses PortAudio (default capture device) to collect voice samples
+
+use Audio::PortAudio;
+use Math::Round qw/round/;
+use LWP::UserAgent;
+use IO::Handle;
+use JSON;
+use Statistics::Basic qw(:all);
+use IO::Handle;
+use MIME::Base64;
+use File::Path qw(make_path);
+
+$|++;
+
+# Audio format
+use constant RATE => 16000;
+use constant NUMCHANNELS   => 1;
+use constant BITSPERSAMPLE => 16;
+
+# Calculate number of samples per chunk based on a given chunk size in milliseconds
+use constant CHUNK_SIZE_MS => 20;
+use constant SAMPLES => RATE * CHUNK_SIZE_MS / 1000;
+
+# Miniumum number of non-silent chunks to count as utterance. Anything less is noise
+use constant MIN_SPEECH => 5;
+
+# Detect 500ms silence (25 blocks * 20 ms ~500ms of speech) before termiating recording
+use constant TRAILING_SILENCE_BLOCKS => 25;
+
+# Depth of FIFO buffer in blocks
+use constant FIFO_DEPTH => 25;
+
+use constant LANG => 'en';
+
+# KITT AI API
+use constant APITOKEN => 'PUT_YOUR_TOKEN_HERE';
+
+# REST endpoint for model training
+use constant URL => 'https://snowboy.kitt.ai/api/v1/train/';
+
+# Storage for generated models / samples
+$models = '/var/spool/models';
```

Comment:
since /tmp gets wiped outon most linux dists after reboot - can we use user's home directory to store models?